api/core/model_runtime/model_providers/voyage/text_embedding/text_embedding.py

```diff
@@ -0,0 +1,174 @@
+import time
+from json import JSONDecodeError, dumps
+from typing import Optional
+
+import requests
+
+from core.embedding.embedding_constant import EmbeddingInputType
+from core.model_runtime.entities.common_entities import I18nObject
+from core.model_runtime.entities.model_entities import AIModelEntity, FetchFrom, ModelPropertyKey, ModelType, PriceType
+from core.model_runtime.entities.text_embedding_entities import EmbeddingUsage, TextEmbeddingResult
+from core.model_runtime.errors.invoke import (
+    InvokeAuthorizationError,
+    InvokeBadRequestError,
+    InvokeConnectionError,
+    InvokeError,
+    InvokeRateLimitError,
+    InvokeServerUnavailableError,
+)
+from core.model_runtime.errors.validate import CredentialsValidateFailedError
+from core.model_runtime.model_providers.__base.text_embedding_model import TextEmbeddingModel
+from core.model_runtime.model_providers.voyage.text_embedding.voyage_tokenizer import VoyageTokenizer
+
+
+class VoyageTextEmbeddingModel(TextEmbeddingModel):
+    """
+    Model class for Voyage text embedding model.
+    """
+
+    api_base: str = "https://api.voyageai.com/v1"
+
+    def _invoke(
+        self,
+        model: str,
+        credentials: dict,
+        texts: list[str],
+        user: Optional[str] = None,
+        input_type: EmbeddingInputType = EmbeddingInputType.DOCUMENT,
```

Comment:
for voyage embedding model,  it  has supported `input_type` for different embedding task. Otherwise, its effectiveness will be greatly reduced. 
it's document below :
<img width="955" alt="image" src="https://github.com/user-attachments/assets/f9fa60d7-4627-4cce-93e6-31b6ac6389b7">

Reply:
@JohnJyong I have add input_type for text_embedding, pls help me to review again, thanks.